optimum/neuron/trainers.py

```diff
@@ -1465,3 +1504,444 @@ class Seq2SeqNeuronTrainer(AugmentTrainerForNeuronMixin, Seq2SeqTrainer):
+        model_init_kwargs: Optional[Dict] = None,
+        dataset_kwargs: Optional[Dict] = None,
+        eval_packing: Optional[bool] = None,
+    ):
+        if not is_trl_available():
+            raise RuntimeError("Using NeuronSFTTrainer requires the trl library.")
+
+        from trl.extras.dataset_formatting import get_formatting_func_from_dataset
+
+        # This will be changed to :
+        # from trl.trainer.callbacks import RichProgressCallback
+        # In the next release.
+        from trl.trainer.utils import (
+            DataCollatorForCompletionOnlyLM,
+            RichProgressCallback,
+            peft_module_casting_to_bf16,
+        )
+
+        if is_peft_available():
+            from peft import PeftConfig, prepare_model_for_kbit_training
+
+        if args is None:
+            output_dir = "tmp_trainer"
+            warnings.warn(f"No `SFTConfig` passed, using `output_dir={output_dir}`.")
+            args = NeuronSFTConfig(output_dir=output_dir)
+        elif args is not None and args.__class__.__name__ == "NeuronTrainingArguments":
+            args_as_dict = args.to_dict()
+            # Manually copy token values as TrainingArguments.to_dict() redacts them
+            args_as_dict.update({k: getattr(args, k) for k in args_as_dict.keys() if k.endswith("_token")})
+            args = NeuronSFTConfig(**args_as_dict)
+
+        if model_init_kwargs is not None:
+            warnings.warn(
+                "You passed `model_init_kwargs` to the SFTTrainer, the value you passed will override the one in the `SFTConfig`."
+            )
+            args.model_init_kwargs = model_init_kwargs
+        if getattr(args, "model_init_kwargs", None) is None:
+            model_init_kwargs = {}
+        elif not isinstance(model, str):
+            raise ValueError("You passed model_init_kwargs to the SFTConfig, but your model is already instantiated.")
+        else:
+            model_init_kwargs = args.model_init_kwargs
+            torch_dtype = model_init_kwargs.get("torch_dtype")
+            if torch_dtype is not None:
+                # Convert to `torch.dtype` if an str is passed
+                if isinstance(torch_dtype, str) and torch_dtype != "auto":
+                    torch_dtype = getattr(torch, torch_dtype)
+                if torch_dtype != "auto" and not isinstance(torch_dtype, torch.dtype):
+                    raise ValueError(
+                        f"Invalid `torch_dtype` passed to the SFTConfig. Expected a string with either `torch.dtype` or 'auto', but got {torch_dtype}."
+                    )
+                model_init_kwargs["torch_dtype"] = torch_dtype
+
+        if infinite is not None:
+            warnings.warn(
+                "The `infinite` argument is deprecated and will be removed in a future version of TRL. Use `TrainingArguments.max_steps` or `TrainingArguments.num_train_epochs` instead to control training length."
+            )
```

Comment:
I like this kind of warning, I usually just remove it to keep less code and avoid users from using deprecated args.

Reply:
Done as well.